pkg/api/types.go

```diff
@@ -2070,10 +2070,41 @@ const (
 	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
 )
 
+// Type and constants for component type.
+type ComponentType string
+
+// These are the valid conditions for the component type.
+const (
+	ComponentAPIServer         ComponentType = "apiserver"
+	ComponentControllerManager ComponentType = "controller-manager"
```

Comment:
When you do a "create" it generates a name like `scheduler-12345` based on the Type. It returns this name so that the caller can use it as a unique identifier, for updates.